File: 2456-construct-smallest-number-from-di-string/2456-construct-smallest-number-from-di-string.cpp
class Solution {
public:
    string smallestNumber(string p) {
        stack<int> st;
        int n = p.size();
        string s = "";

        for (int i = 0; i <= n; i++) {

            st.push(i + 1);

            if (i == n || p[i] == 'I') {
                while (!st.empty()) {
                    s += to_string(st.top());
                    st.pop();
                }
            }
        }
        return s;
    }
};

File: 2456-construct-smallest-number-from-di-string/2456-construct-smallest-number-from-di-string.java
class Solution {
    public String smallestNumber(String p) {
        Stack<Integer> st = new Stack<>();
        StringBuilder s = new StringBuilder();
        int n = p.length();

        for (int i = 0; i <= n; i++) {

            st.push(i + 1);

            if (i == n || p.charAt(i) == 'I') {
                while (!st.empty()) {
                    s.append(st.pop());
                }

            }
           // System.out.println(i);
        }

        return s.toString();
    }
}